dev/benchmarks/macrobenchmarks/lib/src/web/recorder.dart

```diff
@@ -1182,3 +1188,76 @@ void _dispatchEngineBenchmarkValue(String name, double value) {
     listener(value);
   }
 }
+
+abstract class EmptyRecorder extends Recorder implements FrameRecorder {
+  EmptyRecorder({
+    @required String name,
+    this.useCustomWarmUp = false,
+  }) : super._(name, true);
+
+  /// Creates a widget to be benchmarked.
+  ///
+  /// The widget must create its own animation to drive the benchmark. The
+  /// animation should continue indefinitely. The benchmark harness will stop
+  /// pumping frames automatically.
+  Widget createWidget();
+
+  final List<VoidCallback> _didStops = <VoidCallback>[];
```

Comment:
nit: rename to `_didStopCallbacks`.

For consistency, `WidgetRecorder` should also support multiple callbacks, but perhaps making this class a super-class of `WidgetRecorder` is a good way to gain both consistency and code reuse.

Reply:
Changed the name.
I don't understand "WidgetRecorder should also support multiple callbacks" (since it does support) and I'll assume it's no longer valid after removing `EmptyRecorder`.